pkg/db/src/implementations/version-1.2/customer-orders.ts

```diff
@@ -0,0 +1,168 @@
+import { BehaviorSubject, firstValueFrom, Observable, ReplaySubject, share, Subject } from "rxjs";
+
+import { VolumeStock } from "@librocco/shared";
+
+import { DocType } from "@/enums";
+
+import { VersionedString, OrdersDatabaseInterface, CustomerOrderInterface, CustomerOrderData } from "@/types";
+
+import { versionId } from "./utils";
+import { isEmpty, isVersioned, runAfterCondition, uniqueTimestamp } from "@/utils/misc";
+import { newDocumentStream } from "@/utils/pouchdb";
+
+
+class CustomerOrder implements CustomerOrderInterface {
+
+	// {
+	// 	"id": string, // use 'timestampedId' (or 'timestampId', not sure, it's in the 'db' package, under utils I think
+	// 	"draft": boolean,
+	// 	"displayName": string, // "name" -> "displayName" to keep consistent with the rest of the app
+	// 	"email": string,
+	// 	"books": [ 
+	// 	  { isbn: string }, // No status for now we can add that later when dealing with supplier orders
+	// 	],
+	// 	"deposit"?: number // I think the deposit should be calculated on display without persisting and then added when the order is placed
+	// 	}
+
+	#db: OrdersDatabaseInterface;
+
+	#initialized = new BehaviorSubject(false);
+	#exists = false;
+
+	draft = true;
+
+	orderId = "";
+	email = "";
+	deposit = 0;
+
+	books = []
+
+	// Update stream receives the latest document (update) stream from the db. It's multicasted using plain RxJS Subject (no repeat or anything).
+	// This stream is used to signal the update has happened (and has been streamed to update the instance). Subscribers needing to be notified when
+	// an update happens should subscribe to this stream.
+	#updateStream: Observable<CustomerOrderData>;
+	// The stream is piped from the update stream, only it's multicasted using a ReplaySubject, which will cache the last value emitted by the stream,
+	// for all new subscribers. Subscribers needing the latest (up-to-date) data and not needing to be notified when the NEXT update happened, should
+	// subscribe to this stream.
+	#stream: Observable<CustomerOrderData>;
+
+	// timestramped uid
+	_id: VersionedString;
+	docType = DocType.CustomerOrder;
+	_rev?: string;
+	_deleted?: boolean;
+
+
+	entries: VolumeStock[] = [];
+	committed = false;
+	displayName = "";
+	updatedAt: string | null = null;
```

Comment:
This seems to be a leftover from when copying the (note) code from the inventory db.